common/util/hal_snoop.c

```diff
@@ -92,3 +108,63 @@ void snoop_start_thread(){
 void snoop_abort_thread(){
   k_thread_abort(snoop_tid);
 }
+
+void send_post_code_to_BMC() {
+  int send_start_position = 0, send_end_position;
+  ipmi_msg *send_postcode_msg;
+  ipmb_error status;
+
+  while (1) {
+    k_msleep(100); // send post code to BMC once 100 ms
+    send_end_position = snoop_read_num;
+    if (get_DC_status() == 0) {
+      return;
+    }
+    if (send_start_position != send_end_position) {
+      send_postcode_msg = (ipmi_msg*)malloc(sizeof(ipmi_msg));
+      if (send_postcode_msg == NULL) {
+        printk("send post code thread alloc fail\n");
+        continue;
```

Comment:
Can you add a retry count to this malloc failure and error out with a message if exceeded?  Otherwise, this could turn into an infinite loop.

Reply:
We think this thread only works on sending post code in this while loop with 100ms sleep and it shouldn't be terminated before post complete.
So we would like to add retry count after post complete.
Is that make sense to you?

Reply:
Yes, that makes sense.